packages/alfa-style/src/property/font.ts

```diff
@@ -0,0 +1,237 @@
+    >;
+
+    export type Relative = Keyword<"larger" | "smaller">;
+  }
+
+  /**
+   * Base font sizes for generic font families. In browsers, these sizes depend
+   * on a number of things, such as language and user configuration, but for
+   * simplicity we assume a single base size for every generic font family.
+   */
+  const bases: { [P in Family.Generic["value"]]: number } = {
+    serif: 16,
+    "sans-serif": 16,
+    monospace: 13,
+    cursive: 16,
+    fantasy: 16
+  };
+
+  /**
+   * Scaling factors for absolute font sizes. In browsers, these factors are
+   * only used when the base font size are outside the range of 9 to 16 pixels.
+   * Within this range, a hardcoded lookup table is used for various legacy
+   * reasons but for simplicity we always rely on these scaling factors. The
+   * effect of this is that we will compute font sizes that are slightly larger
+   * than what browsers would render.
+   *
+   * @see https://drafts.csswg.org/css-fonts/#absolute-size-mapping
+   */
+  const factors: { [P in Size.Absolute["value"]]: number } = {
+    "xx-small": 3 / 5,
+    "x-small": 3 / 4,
+    small: 8 / 9,
+    medium: 1,
+    large: 6 / 5,
+    "x-large": 3 / 2,
+    "xx-large": 2 / 1,
+    "xxx-large": 3 / 1
+  };
+
+  /**
+   * @see https://drafts.csswg.org/css-fonts/#propdef-font-size
+   */
+  export const Size: Property<Size, Size.Computed> = Property.of(
+    Length.of(16, "px"),
+    either(
+      either(
+        Keyword.parse(
+          "xx-small",
+          "x-small",
+          "small",
+          "medium",
+          "large",
+          "x-large",
+          "xx-large",
+          "xxx-large"
+        ),
+        Keyword.parse("larger", "smaller")
+      ),
+      either(Percentage.parse, Length.parse)
+    ),
+    style =>
+      style.specified("font-size").map(size => {
+        switch (size.type) {
+          case "length":
+            return Resolver.length(size, style.parent);
+
+          case "percentage": {
+            const parent = style.parent.computed("font-size").value;
+
+            return Length.of(parent.value * size.value, parent.unit);
+          }
+
+          case "keyword": {
+            const parent = style.parent.computed("font-size").value;
+
+            switch (size.value) {
+              case "larger":
+                return Length.of(parent.value * 1.2, parent.unit);
+
+              case "smaller":
+                return Length.of(parent.value * 0.85, parent.unit);
+
+              default: {
+                const factor = factors[size.value];
+
+                const [family] = style.computed("font-family").value;
+
+                const base =
+                  family.type === "keyword" ? bases[family.value] : bases.serif;
+
+                return Length.of(factor * base, "px");
+              }
+            }
+          }
+        }
+      }),
+    {
+      inherits: true
+    }
+  );
+
+  export type Weight = Weight.Asbolute | Weight.Relative | Number;
```

Comment:
```suggestion
  export type Weight = Weight.Computed | Weight.Absolute | Weight.Relative;
```
Keeps both the symmetry in the names (`Weight.*`) and the same order as in the namespace definition below, making it easier to visually debug.

(+ Asbolute => Absolute)

Reply:
The reason why `Computed` is a type of its own and not used in the initial definition of `Weight` is that it's not a given that a CSS property computes to something that can also be specified. As such, the pattern across properties will be:

```ts
type Foo = string | number;

namespace Foo {
  type Computed = number;
}

const Foo: Property<Foo, Foo.Computed> = ...;
```

Come to think of it, it may be better to be explicit and consistently declare a `Specified` type instead:

```ts
namespace Foo {
  type Specified = string | number;
  type Computed = number;
}

const Foo: Property<Foo.Specified, Foo.Computed>;
```

If a given CSS property computes to its specified value, we can either choose to be explicit about it:

```ts
namespace Foo {
  type Specified = string | number;
  type Computed = Specified;
}

const Foo: Property<Foo.Specified, Foo.Computed>;
```

...or leave out the type altogether:

```ts
namespace Foo {
  type Specified = string | number;
}

const Foo: Property<Foo.Specified>;
```

Thoughts?

Reply:
Hum… The visual weirdness here come from the pattern
```javascript
type Foo = Foo.Specified | Number
namespace Foo {
 export type Computed = Number;
 export type Specified = Keywords<…>
}
```

---
I'm rather in favour of
```javascript
namespace Foo {
  export type Specified = … /* can be several such as Absolute and Relative */
  export type Computed = … /* explicit "Specified" if needed */
}
const Foo: Property<Foo.Specified, Foo.Computed> /* Always. Nice pattern. */
```

---
But if we need to have an external `type Foo` I think its pattern should be
```javascript
type Foo = Foo.Specified | Foo.Computed
```
rather than 
```javascript
type Foo = Foo.Specified | magic-value-that-happens-to-be-the-same-as-Foo.Computed
```

Here, we do have a `Foo.Specified` (actually 2) and thus the `type Foo` should used **both** `Foo.Specified` and `Foo.Computed` rather than mixing `Foo.Specified` and a magic value that can fall out of sync.